cdap-docs/admin-manual/source/operations/logging.rst

```diff
@@ -1,77 +1,573 @@
+Log Saver Service
+-----------------
+These :ref:`properties <appendix-cdap-default-logging>` control the Log Saver Service:
 
-:term:`Master Services <master services>` use ``logback-container.xml``, packaged with the CDAP distribution,
-for logging configuration. This logback does log rotation once a day at midnight and expires logs older than
-14 days. Changes can be made to ``logback-container.xml``, and the ``cdap-master`` process will need to be restarted
-for the modified logback to take effect.
+.. generated by code in conf.py for this manual
+.. include:: /../target/_includes/logging-log-saver-service.rst
+
+Log saver instances should be from a minimum of one to a maximum of ten. The maximum is
+set by the number of Kafka partitions (``log.publish.num.partitions``), which by default is 10.
+
+.. _logging-monitoring-log-pipeline-configuration:
+
+Log Pipeline Configuration
+--------------------------
+Configuration properties for logging and :ref:`custom log pipelines
+<logging-monitoring-configuring-custom-log-pipelines>` are shown in the documentation of the
+:ref:`logging properties <appendix-cdap-default-logging>` section of the
+:ref:`cdap-site.xml <appendix-cdap-site.xml>` file.
+
+The **CDAP log pipeline** is configured by settings in the :ref:`cdap-site.xml
+<appendix-cdap-site.xml>` file.
+
+**Custom log pipelines** are configured by a combination of the settings in the 
+:ref:`cdap-site.xml <appendix-cdap-site.xml>` file and a "logback" file used
+to specify the custom pipeline. The XML file is placed in the
+``log.process.pipeline.config.dir``, a local directory on the CDAP Master node that is scanned
+for log processing pipeline configurations. Each pipeline is defined by a file in the
+Logback XML format, with ``.xml`` as the file name extension.
+
+These :ref:`properties <appendix-cdap-default-logging>` control the CDAP log pipeline:
+
+.. generated by code in conf.py for this manual
+.. include:: /../target/_includes/logging-pipeline-configuration-1.rst
+
+.. _logging-monitoring-cdap-site-xml-configuration:
+
+These :ref:`properties <appendix-cdap-default-logging>` control both the CDAP log pipeline
+and custom log pipelines:
+
+.. generated by code in conf.py for this manual
+.. include:: /../target/_includes/logging-pipeline-configuration-2.rst
+
+The ``log.process.pipeline.*`` properties can be over-ridden and specified at the
+custom pipeline level by providing a value in a pipeline's "logback" file for any of
+these properties.
+
+
+Logging Framework
+=================
+
+.. highlight:: console
+
+This diagram shows in greater detail the components and steps CDAP follows when logging
+programs of an application and system services with the logging framework:
+
+  - Logs are collected from individual programs running in YARN containers.
+
+  - YARN writes the log messages emitted by containers to files inside the container.
+
+  - CDAP System Services run (depending on the service) either on cluster edge nodes or in
```

Comment:
flip 3rd and 4th bullet. The "these messages" really refers to the same log messages that Yarn writes.